util/OptionsDB.cpp

```diff
@@ -48,6 +49,79 @@ namespace {
         return str.substr(1, str.size() - 2);
     }
 
+    template<typename T>
+    void AppendNativeCharToXMLText(std::string& text, T ch) noexcept {
+        const uint_fast32_t chval = static_cast<uint_fast32_t>(static_cast<std::make_unsigned_t<T>>(ch));
+        switch (chval) {
+        case static_cast<uint_fast32_t>('&'):
+            text.append("&amp;");
+            break;
+        case static_cast<uint_fast32_t>('<'):
+            text.append("&lt;");
+            break;
+        case static_cast<uint_fast32_t>('>'):
+            text.append("&gt;");
+            break;
+        case static_cast<uint_fast32_t>('\''):
+            text.append("&apos;");
+            break;
+        case static_cast<uint_fast32_t>('\"'):
+            text.append("&quot;");
+            break;
+        default:
+            if (chval >= 20 && chval <= 127) {
+                text.push_back(static_cast<char>(chval));
+            } else {
+                text.append("&#").append(std::to_string(chval)).append(";");
+            }
+        }
+    }
+
+    template<typename T>
+    std::basic_string<T> ConvertXMLTextToPath(std::string_view input_string) noexcept {
+        std::basic_string<T> retval{};
+        for (size_t index = 0; index < input_string.size(); ++index) {
+            const auto ch = input_string[index];
+            if (ch == '&') {
+                const auto semicolon_index = input_string.find(';', index);
+                if (semicolon_index == std::string_view::npos) {
+                    ErrorLogger() << "Cann't find semicolon after ampersand in string: " << input_string;
+                    retval.push_back('?');
+                    return retval;                
+                }
+                const auto entity = input_string.substr(index + 1, semicolon_index - index - 1);
+                if (entity == "amp") {
+                    retval.push_back('&');
+                } else if (entity == "lt") {
+                    retval.push_back('<');
+                } else if (entity == "gt") {
+                    retval.push_back('>');
+                } else if (entity == "apos") {
+                    retval.push_back('\'');
+                } else if (entity == "quot") {
+                    retval.push_back('\"');
+                } else if (entity.starts_with('#')) {
+                    try {
+                        const auto number = entity.substr(1);
+                        uint_fast32_t code;
+                        std::from_chars(number.data(), number.data() + number.size(), code);
```

Comment:
not sure if we always have `from_chars` available yet... there's some `#if defined(__cpp_lib_to_chars)` with fallback to boost lexical cast... make sure this works for all test cases?

Reply:
probably need to check the return value from `from_chars` rather than (or also?) catching an exception. cppreference says `from_chars` doesn't throw.